delegation-verify-coordinator/templates/role_job.yaml

```diff
@@ -3,9 +3,8 @@ apiVersion: rbac.authorization.k8s.io/v1
 kind: Role
 metadata:
   namespace: {{ .Release.Namespace }}
-  name: {{ .Release.Name }}
+  name: {{ .Release.Name }}-job
```

Comment:
the `-job` suffix, is confusing as this role is applied to a kind `deployment`, not a kind `job`